docs/samples/Microsoft.ML.Samples/Dynamic/Transforms/ImageAnalytics/ConvertToImage.cs

```diff
@@ -32,35 +33,39 @@ public static void Example()
 
             var transformedData = pipeline.Fit(data).Transform(data);
 
-            // Preview the transformedData. 
-            var transformedDataPreview = transformedData.Preview();
-            PrintPreview(transformedDataPreview);
+            // Preview the transformedData.
+            PrintColumns(transformedData);
+
             // Features                 Image                    Pixels
-            // 185,209,196,142,52       System.Drawing.Bitmap    185,209,196,142,52
-            // 182,235,84,23,87         System.Drawing.Bitmap    182,235,84,23,87
-            // 192,214,247,22,38        System.Drawing.Bitmap    192,214,247,22,38
-            // 242,161,141,223,192      System.Drawing.Bitmap    242,161,141,223,192
+            // 185,209,196,142,52...    {Width=224, Height=224}  185,209,196,142,52...
+            // 182,235,84,23,87...      {Width=224, Height=224}  182,235,84,23,87...
+            // 192,214,247,22,38...     {Width=224, Height=224}  192,214,247,22,38...
+            // 242,161,141,223,192...   {Width=224, Height=224}  242,161,141,223,192...
         }
 
-        private static void PrintPreview(DataDebuggerPreview data)
+        private static void PrintColumns(IDataView transformedData)
         {
-            foreach (var colInfo in data.ColumnView)
-                Console.Write("{0,-25}", colInfo.Column.Name);
-
-            Console.WriteLine();
-            foreach (var row in data.RowView)
+            Console.WriteLine("{0, -25} {1, -25} {2, -25}", "Features", "Image", "Pixels");
+            using (var cursor = transformedData.GetRowCursor(transformedData.Schema))
             {
-                foreach (var kvPair in row.Values)
+                var featuresGetter = cursor.GetGetter<VBuffer<float>>(cursor.Schema["Features"]);
+                var pixelsGetter = cursor.GetGetter<VBuffer<float>>(cursor.Schema["Pixels"]);
+                var imageGetter = cursor.GetGetter<Bitmap>(cursor.Schema["Image"]);
+                while (cursor.MoveNext())
                 {
-                    if (kvPair.Key == "Pixels" || kvPair.Key == "Features")
-                    {
-                        var rawValues = ((VBuffer<float>)kvPair.Value).DenseValues().Take(5);
-                        Console.Write("{0,-25}", string.Join(",", rawValues));
-                    }
-                    else
-                        Console.Write("{0,-25}", kvPair.Value);
+                    VBuffer<float> features = default;
+                    featuresGetter(ref features);
```

Comment:
So, here's where my note becomes quite practically important. You've broken buffer sharing on `VBuffer`, by putting the declaration inside the while loop. So you're forcing reallocation, a big no-no.

Try [this documentation](https://github.com/dotnet/machinelearning/blob/master/docs/code/VBufferCareFeeding.md#buffer-re-use-as-a-user), I also talk about this principle here. Good idea to know how to use `IDataView` correctly.

Reply:
sounds good. May be will put a comment as well to educate the users as they wouldn't know this away.

Reply:
Certainly can't hurt to reinforce the lesson, since we directly use `IDataView` in relatively few samples.

---
In reply to: [275874929](https://github.com/dotnet/machinelearning/pull/3307#discussion_r275874929) [](ancestors = 275874929)